components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationManagerImpl.java

```diff
@@ -541,6 +542,23 @@ public String getPrimaryOrganizationId(String organizationId) throws Organizatio
         return null;
     }
 
+    @Override
+    public Organization addRootOrganization(int tenantId, Organization organization)
+            throws OrganizationManagementException {
+
+        try {
+            setCreatedAndLastModifiedTime(organization);
+            organization.setType(TENANT.name());
+            organizationManagementDAO.addRootOrganization(organization);
+            org.wso2.carbon.user.api.Tenant tenant = getRealmService().getTenantManager().getTenant(tenantId);
+            tenant.setAssociatedOrganizationUUID(organization.getId());
+            getRealmService().getTenantManager().updateTenant(tenant);
+        } catch (UserStoreException e) {
+            throw handleServerException(ERROR_CODE_ERROR_CREATING_ROOT_ORGANIZATION, e, String.valueOf(tenantId));
```

Comment:
on any failure of adding root org, shall we deactivate the tenant?
Otherwise, the tenant will fail to use the organization mgt feature
Also, some features might not work on failure to resolve the org id. (eg: rolev2 audience can be organization id)

Reply:
Fixed by - https://github.com/wso2/identity-organization-management-core/pull/89/commits/9c30415a2ea1cfc93bd23a06f09254730216aee9

The tenant was deactivated if root organization creation fails.